src/ArborX_CrsGraphWrapper.hpp

```diff
@@ -26,17 +26,8 @@ inline void query(Tree const &tree, ExecutionSpace const &space,
 {
   Kokkos::Profiling::pushRegion("ArborX::query");
 
-  Details::CrsGraphWrapperImpl::
-      check_valid_callback_if_first_argument_is_not_a_view(callback_or_view,
-                                                           predicates, view);
-
-  using Access = AccessTraits<Predicates, ArborX::PredicatesTag>;
-  using Tag = typename Details::AccessTraitsHelper<Access>::tag;
-
-  ArborX::Details::CrsGraphWrapperImpl::queryDispatch(
-      Tag{}, tree, space, predicates,
-      std::forward<CallbackOrView>(callback_or_view), std::forward<View>(view),
-      std::forward<Args>(args)...);
+  tree.query(space, predicates, std::forward<CallbackOrView>(callback_or_view),
+             std::forward<View>(view), std::forward<Args>(args)...);
```

Comment:
I am not sure about pushing a profiling region here.
I think this should read something like
```C++
template <typename Tree, typename ExecutionSpace, typename Predicates,
          typename... Args>
void query(Tree const &tree, ExecutionSpace const &space,
           Predicates const &predicates, Args &&...args) {
  tree.query(space, predicates, std::forward<Args>(args)...);
}
```
potentially asserting that ExecutionSpace is a proper Kokkos execution space or whatnot...